File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/managetaskview/TaskViewModel.java
package com.wiseassblog.samsaradayplanner.ui.managetaskview;

import com.wiseassblog.samsaradayplanner.domain.Task;

/**
 * A data store (cache) for front end session data
 */
public class TaskViewModel implements ITaskViewContract.ViewModel {
    private Task task;

    @Override
    public void setTask(Task task) {
        this.task = task;
    }

    @Override
    public Task getTask() {
        return this.task;
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/managetaskview/TaskViewModel.kt
package com.wiseassblog.samsaradayplanner.ui.managetaskview

import com.wiseassblog.samsaradayplanner.domain.Task

/**
 * A data store (cache) for front end session data
 */
class TaskViewModel : ITaskViewContract.ViewModel {
    override lateinit var task: Task
}